### PR TITLE
add arrayAssertions for the new infix API

### PR DESCRIPTION
**Pull request by isfedorov**
This PR addresses issue #226. @robstoll I have a couple of questions: 1) `change since 0.9.0 to 0.11.0` but currently there is no any `since` version. Should I just add `since 0.11.0` to every function added to `arrayAssertions`? 2) I'm not sure that I've properly updated infix methods to use filler object `o`.  Could you please check it? Also I've left description as is from fluent arrayAssertions. Please notify me if descriptions should be updated.
_______________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/ArrayAsListAssertionsSpec.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        fun charArray(plant: Expect<CharArray>) = plant asList o
+        fun shortArray(plant: Expect<ShortArray>) = plant asList o
+        fun intArray(plant: Expect<IntArray>) = plant asList o
+        fun longArray(plant: Expect<LongArray>) = plant asList o
+        fun floatArray(plant: Expect<FloatArray>) = plant asList o
+        fun doubleArray(plant: Expect<DoubleArray>) = plant asList o
+        fun booleanArray(plant: Expect<BooleanArray>) = plant asList o
+
+        fun arrayIntWithCreator(plant: Expect<Array<Int>>, assertionCreator: Expect<List<Int>>.() -> Unit) =
+            plant asList assertionCreator
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/ArrayAsListAssertionsSpec.kt:41]: please change this line to (not that yours is wrong, I just like to see the syntax as it is used afterwards)
```suggestion
            plant asList { assertionCreator() }
```

**Comment by robstoll** [comment]
about the `@since 0.11.0`: please add it to every function. Would you mind to add `@since 0.9.0` to the fluent API? Seems like we missed it there.

**Comment by isfedorov** [comment]
@robstoll seems I've resolved all the comments you left. So I'm removing [WIP] from the title but if you'll find anything else that should be fixed please ping me

**Comment by robstoll** [comment]
@isfedorov perfect, thanks :+1:

**Pull request merged**